src/java_bytecode/java_object_factory.cpp

```diff
@@ -1028,6 +1030,25 @@ void java_object_factoryt::gen_nondet_struct_init(
         substruct_in_place);
     }
   }
+
+  // If <class_identifier>.cproverValidate() can be found in the
+  // symbol table, we add a call:
+  // ~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+  // expr.cproverValidate();
+  // ~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+  const irep_idt function_name =
+    "java::" + as_string(class_identifier) + ".cproverValidate:()V";
+  if(symbol_table.has_symbol(function_name))
```

Comment:
Note we'll need to do something about this when lazy loading improves -- right now because the rule is "instantiation site + call site == function callable," this sneaks through the weaknesses of lazy-loading-v1. Future lazy loading will be cleverer and will need a rule to always check for and load these.